pkg/registry/pod/rest.go

```diff
@@ -55,6 +55,9 @@ func (podStrategy) PrepareForCreate(obj runtime.Object) {
 	pod.Status = api.PodStatus{
 		Phase: api.PodPending,
 	}
+	for i := range pod.Spec.ImagePullSecrets {
+		pod.Spec.ImagePullSecrets[i] = api.ObjectReference{Name: pod.Spec.ImagePullSecrets[i].Name}
```

Comment:
Since this is a new feature, can we just do that now? As explained to me by @bgrant0607, the pattern would be to make the field ```ImagePullSecrets``` into ```ImagePullSecretNames []string```, to later be replaced by a ```ImagePullSecrets []TBD``` where TBD is a specific name referencing type.